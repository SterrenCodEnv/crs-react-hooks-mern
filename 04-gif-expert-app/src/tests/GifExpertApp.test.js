import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";

describe("Test in GifExpertApp", () => {
  test("Show GifExpertApp component correctly", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Show categories list", () => {
    const categories = ["The Office", "The Big Bang Theory"];
    const wrapper = shallow(<GifExpertApp defaultCategory={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
