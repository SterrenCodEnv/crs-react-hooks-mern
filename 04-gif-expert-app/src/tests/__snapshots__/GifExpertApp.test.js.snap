// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp Show GifExpertApp component correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Test in GifExpertApp Show categories list 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The Office"
      key="The Office"
    />
    <GifGrid
      category="The Big Bang Theory"
      key="The Big Bang Theory"
    />
  </ol>
</div>
`;
